.github/workflows/tests.yml

```diff
@@ -0,0 +1,89 @@
+name: CI tests
+
+defaults:
+  run:
+    shell: bash
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+
+    container: 
+        image: koreader/koappimage:0.2.2
+        options: --user ko
+        
+    env:
+        EMULATE_READER: 1
+        # this is for shellcheck 0.4.5 and lower; can be removed for 0.4.6
+        LC_ALL: en_US.UTF8
+        HOME: /home/ko
+        # pretend to be cirleci so the scripts dont have to change
+        CIRCLE_WORKING_DIRECTORY: ${{ github.workspace }}
+
+    steps:
+        # chown working directory to ko (docker user0
+      - name: permissions
+        run: sudo chown -R ko:ko /__w/koreader/
+
+        # make sure we have git >= 2.18 so github checkout behaves
+      - name: git update
+        run: sudo add-apt-repository ppa:git-core/ppa -y && sudo apt-get update && sudo apt-get install git -y
+
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Cache deps
+        uses: actions/cache@v2
+        with:
+            path: |
+                /home/ko/.ccache
+                base
+            key: deps-${{ runner.os }}-${{ hashFiles('.ci/install.sh') }}-${{ hashFiles('.ci/helper_luarocks.sh') }}
+
+      - name: need to init some stuff first or git will complain when sticking in base cache
+        run: git submodule init base && git submodule update base && pushd base && git submodule init && git submodule update && popd
+
+      - name: cache check
+        run: pushd base && git_rev_base=$(git describe HEAD) && popd && echo $git_rev_base && echo $git_rev_base >git-rev-base
+
+      - name: Cache base
+        uses: actions/cache@v2
+        with:
+            path: |
+                /home/ko/bin
```

Comment:
The separate shellcheck and shfmt downloads were really more of a thing from back in Travis before we used Docker. I just integrated them in the image in part because I noticed they were outdated in base. That being said, it could be useful to maintain the ability to easily override without having to create a whole new Docker image. https://github.com/koreader/virdevenv/pull/68

Reply:
are you referring to `install.sh`?